developer-docs/bolt_server.md

```diff
@@ -0,0 +1,147 @@
+    ssl-key: /etc/puppetlabs/bolt-server/ssl/private_key.pem
+    ssl-ca-cert: /etc/puppetlabs/bolt-server/ssl/ca.pem
+} 
+```
+
+## API Endpoints
+Each API endpoint accepts a request as described below. The request body must be a JSON object.
+
+## POST /ssh/run_task
+- `target`: [SSH Target Object](#ssh-target-object), *required* - Target information to run task on.
+- `task`: [Task Object](#task-object), *required* - Task to run on target.
+- `parameters`: Object, *optional* - JSON formatted parameters to be provided to task.
+
+### SSH Target Object
+The Target be a JSON object. The following keys are available:
+- `hostname`: String, *required* - Target identifier.
+- `password`: String, *optional* - Password for SSH transport authentication.
+- `port`: Integer, *optional* - Connection port (Default: `22`).
+- `user`: String, *optional* - Login user (Default: `root`).
+- `connect-timeout`: Integer, *optional* - How long Bolt should wait when establishing connections.
+- `run-as-command`: Array, *optional* - Command elevate permissions. Bolt appends the user and command strings to the configured `run-as` command before running it on the target. This command must not require an interactive password prompt, and the `sudo-password` option is ignored when `run-as-command` is specified.
+- `run-as`: String, *optional* - A different user to run commands as after login.
+- `tmpdir`: String, *optional* - The directory to upload and execute temporary files on the target.
+- `host-key-check`: Bool, *optional* - Whether to perform host key validation when connecting over SSH (Default: `true`).
+- `sudo-password`: String, *optional* - Password to use when changing users via `run-as`
+
+### Task Object
+The Task be a JSON object. The following keys are available:
+- `name`: String, *required* - The name of the task.
+- `metadata`: String, *optional* - The contents of the task's metadata.json file. 
+- `file`: String, *required* - Task's base64 encoded file content.
+
+## Response 
+If the task runs the response will have status 200.
+The response will be a standard bolt Result JSON object.
+
+## Example
+
+The following example walks through the setup to get the bolt-server running on an Ubuntu 16.04 vmpooler node. 
+
+1. Download and install puppet6 and puppetserver (for certs)
+```
+curl -O http://nightlies.puppet.com/apt/puppet6-nightly-release-xenial.deb
+dpkg -i ./puppet6-nightly-release-xenial.deb 
+apt-get update
+apt-get install puppetserver
+export PATH=$PATH:/opt/puppetlabs/bin
+puppet resource service puppetserver ensure=running
+```
+2. Get certs (Error messages OK here, as long as certs are generated)
+```
+puppet agent -t
+puppet cert list -a
+puppet cert sign [host reported from command above]
+```
+Check that the following certs have been generated
+```
+/etc/puppetlabs/puppet/ssl/certs/$HOSTNAME.pem
+/etc/puppetlabs/puppet/ssl/private_keys/$HOSTNAME.pem
+```
+3. Download and install bolt-server
```

Comment:
Include 'from http://builds.delivery.puppetlabs.net'  since the version in the copy-paste will be out of date, and clicking is easier than figuring out where to go